### DISCUSSION TITLE
A better sidebar UX

### DISCUSSION THREAD

**Discussion by hanaral** [elementary/files#1613] [Ideas]
## Problem

1. There are too many ways to access the same home dir. folders by default. The way users access directories such as Documents or Pictures is duplicated between actual navigation and that provided by bookmarks. 

2. Home, Recent and Trash behave as if they were static buttons, so why are they in the bookmarks list? 

3. I don't expect web browsers to have any bookmarks by default, let alone unremovable ones.

## Proposal

The sidebar should have the home directory, recents and trash as static buttons at the top of the list _(since 2 of them aren't even visible folders)_ and _possibly_ not have any bookmarks by default(?)
(I don't think using the user's tools for them is a good idea, it's just confusing since a bookmark is something you add yourself)

This way you have 3 static buttons to access the 3 base locations:

1. **The home directory**; where user _always_ starts on first launch or new tab. All the default folders are immediately accessible from here anyway
2. **The recents view**;  a collection of recently _used_ items and isn't even a folder
3.  **The trash folder**; a collection of recently _deleted_ items and is only a folder in the literal sense

Another thing that can now be considered is carrying over a bookmarked item/folder's colour tag to the bookmarks

## Prior Art

![Screenshot from 2021-03-03 00-29-16](https://user-images.githubusercontent.com/58219504/109734276-a78a7500-7bb8-11eb-824b-f07ac0295caa.png)

**Comment by pongloongyeat** [comment]
Having them ungrouped just makes it look weird to me and makes the ungrouped list seem redundant (from your pic, there are only 3 buttons).

Fundamentally, I believe you're right (since bookmarks _are_ supposed to be added by the user) but I am in favour of leaving it as it is and besides, most of the default bookmarks can be removed with the exception of home and trash. Perhaps renaming "Bookmarks" would be more appropriate here. Something like "Favourites" or "Places".

**Comment by hanaral** [comment]
I don't think it's appropriate to add bookmarks in this case. The user will always right click files and see that they can do it anyway

**Comment by jeremypw** [comment]
There does seem to be a trend to put user bookmarks in a separate group in other filemanagers, which does seem more logical.  Unfortunately the user bookmark group usually gets put at the bottom of the sidebar which I am :-1: on as it makes them less accessible often requiring scrolling to access. They should be at or near the top. The current layout is probably for historical reasons but people are used to it now - whether there are sufficient downsides to warrant a change is questionable but it would be easy enough to do if the UX team agreed.

**Comment by hanaral** [comment]
What I don't get then is why Files relies on a fundementally different navigation system that is more similar to web browsers - why is it that it is fine to reimagine that, but not something you just said is more logical? I see no problem with reimagining illogical design choices that are there because they were there before.

Anyway, moving 3 buttons that were already there above the category that they aren't in isn't a huge change - argue that there should or should not be bookmarks by default, but it does not change that home, recents and trash are not at all bookmarks.

**Comment by jeremypw** [comment]
I don't disagree.  As the sidebar is currently being rewritten now would be a good time to reconsider past design decisions (which may pre-date elementaryos), but ultimately it is up to the UX team.

**Comment by cassidyjames** [comment]
I think I would be in favor of smaller, more iterative improvements to the sidebar. :) For example, pulling out non-removable items (Recent, Home, Trash from the list of Bookmarks could be a great start (filed as #1614). The whole point of default and user-modifiable bookmarks is ensuring users can get to those things no matter where they are in the file manager or (file open/save dialogs). We set default bookmarks because it's the common/expected places for storing certain types of files, so removing default bookmarks altogether would be a big loss here.

While we have some very web-browser-inspired workflows here, we also have to remember that it's still a fundamentally different experience. The web is a limitless portal of loading third-party experiences; Files is a way to drill down, modify, or quickly access the user's own content. You don't typically organize or experience websites in the same way you organize and "experience" file management, even if we can be heavily inspired by modern workflows of web browsers when addressing browsing a file system. But this is a much larger topic that could be discussed ad infinitum, so I'm gonna stop here. :wink:

**Reply by hanaral** [reply]
Cheers for filing the issue - You're right about the limitless possibilities of the web being quite different _(although you may argue that the current web only exists because of lazy organisations using it as a cheap way to make cross-platform apps)_

It might also be interesting to expand on virtual locations in the sidebar with some of the discussion over in [this issue](https://github.com/elementary/icons/issues/1009#issuecomment-789406973) I might file an issue for it tbh.
